provider/resource_alert_route.go

```diff
@@ -340,6 +347,29 @@ func resourceAlertRouteCreate(ctx context.Context, d *schema.ResourceData, meta
 	d.SetId(res.ID)
 	tflog.Trace(ctx, fmt.Sprintf("created a alert_route resource: %s", d.Id()))
 
+	if requestID != "" {
+		tflog.Debug(ctx, fmt.Sprintf("Waiting for asynchronous rule creation to complete for alert route %s with request_id %s", d.Id(), requestID))
+
+		checkStatusFunc := func(alertRouteID, requestID string) (*polling.AsyncRuleCreationStatus, error) {
```

Comment:
Creating a wrapper function here to avoid passing the client instance to the method defined in alert_route package.